### PR TITLE
fix: topics type for logs in rpc responses

### PR DESCRIPTION
**Pull request by 0xOlias**
AFAICT, the `topics` property of logs in RPC responses can only be an array of hex strings. Elements of this array will never be null, and will never be an array of strings, so I think the type suggested here is more precise. See [this discussion](https://github.com/wagmi-dev/viem/discussions/163) and the [JSON-RPC spec](https://ethereum.github.io/execution-apis/api-documentation/) for more details.

To summarize:
1. The `topics` argument passed to `eth_getLogs` and `eth_newFilter` has the type `(Hex | Hex[] | null)[]`
2. The `topics` property of logs returned by `eth_getLogs`, `eth_getFilterLogs`, and `eth_getFilterChanges` has the type `Hex[]` (and has length <= 4)

### CODE REVIEW

**Review by jxom — COMMENTED** [review]
Can we add a changeset?
